### PR TITLE
Update dependencies to latest version

### PR DESCRIPTION
**Pull request by igorbolic**
Updates the dependencies to latest versions.

This will fix the issue when using `projectreactor` version **3.4.x** where the deprecated retry method:

`Flux<T> retry(long numRetries, Predicate<? super Throwable> retryMatcher)` 

has been removed.

### CODE REVIEW

**Review by roma0297 — APPROVED** [review]

**Review by oleborup — APPROVED** [review]

**Comment by AndrewZurn** [comment]
@mokies - sorry to @ you, would you (or maybe someone else) be able to review/merge and publish this? It looks like there are a handful of people and projects that are likely caught but this subtle API breakage in the underlying lib and can't get the latest release of reactor in their project until this likely is brought in line. Thanks!

**Review by oleborup — APPROVED** [review]

**Review by AndrewZurn — APPROVED** [review]

**Review by Jodee90 — APPROVED** [review]

**Comment by AndrewZurn** [comment]
Checking out twitter the maintainer hasn't tweeted anything since November, so he may be completely out of communication at this point. Maybe it's time a fork is considered?

**Review by mokies — APPROVED** [review]

**Comment by mokies** [comment]
> Checking out the maintainer's twitter, they haven't tweeted anything since November, so they may be completely out of communication at this point. Maybe it's time a fork is considered (or if we can get more maintainers onboarded, which would probably be preferable)?

Apologies to everyone blocked on this issue. Life is pretty busy for me these days and I struggle to find time to maintain this project. 
I'd love to see a couple of maintainers on this project.

**Pull request merged**
